spec/models/spree/order_contents_spec.rb

```diff
@@ -166,6 +166,12 @@
   end
 
   describe "#update_or_create" do
+    it "ensures shipments are updated" do
+      expect(order).to receive(:ensure_updated_shipments)
+
+      subject.update_or_create(variant, { quantity: 2, max_quantity: 3 })
+    end
+
```

Comment:
Looks like this tests creating mode, but not updating an existing line item (Which I think is what it should be testing)

Reply:
I updated the specs to test both in creating and updating scenario